lemur/certificates/cli.py

```diff
@@ -323,10 +323,20 @@ def rotate(endpoint_name, source, new_certificate_name, old_certificate_name, me
                     request_rotation(ep, old_cert, new_cert, message, commit)
                 current_app.logger.info(log_data)
         else:
-            # No certificate name or endpoint is provided. We will now fetch all endpoints,
-            # which are associated with a certificate that has been replaced
-            print("[+] Rotating all endpoints that have new certificates available")
-            for endpoint in endpoint_service.get_all_pending_rotation():
+            # No certificate name or endpoint is provided.
+            # If a source is provided then we rotate endpoints only associated to that source.
+            # Otherwise, we will now fetch all endpoints, which are associated with a certificate that has been replaced
+            source_rotation = False
+            if source is not None or source == "":
```

Comment:
Under what circumstances would it be valid to perform a "rotation by source" if the `--source` flag is empty?

Reply:
good catch. `source == ""` is a bug here. The test didn't catch this because the first check would always return True or False. Looking at this again we can simplify the code further to be
```
if source:
  # get endpoints that need to be rotated from source
else:
  # get all endpoints that need rotation
 ```